### DISCUSSION TITLE
how to use custom functions inside addRoute hook

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#3408] [Q&A] [answered]
here is my code
`function generateUUID(length = 15) {
   return (
    Math.random().toString(36).substring(2) +
     Math.random().toString(36).substring(2)
   ).substring(0, length);
 }

routerAdd("GET", "/t1", (c) => {
  try {
    c.json(200, {
      id:generateUUID();
    });
  } catch (e) {
    c.json(300, {
      message: e.toString()
    });
  }
});
`
output:
{
"message": "ReferenceError: generateUUID is not defined"
}

**Comment by ganigeorgiev** [comment, marked as answer]
You can't call variable and functions defined from outside the handler scope (see [JSVM Caveats and limitations](https://pocketbase.io/docs/js-overview/#caveats-and-limitations) for more details and workaround).

If you want to generate a random string you can use the available [`$security.randomString(15)`](https://pocketbase.io/jsvm/functions/_security.randomString.html)/[`$security.pseudorandomString(15)`](https://pocketbase.io/jsvm/functions/_security.pseudorandomString.html) helpers.
